### PR TITLE
18290 educator slow opening large modules for editing in assign action on ipad

### PR DESCRIPTION
**Pull request by dmiserg**
https://trello.com/c/gTJ1pK9J/18290-educator-slow-opening-large-modules-for-editing-in-assign-action-on-ipad

### CODE REVIEW

File: platforms/common/haxe/HaxeRuntime.hx [hunk 1]
```diff
@@ -341,23 +355,47 @@ if (a === b) return true;
...
+	}
 
-				return "\"" + s + "\"";
-			} else {
-				StringTools.replace(s, "\\", "\\\\"); // Check if really a string
+	public static function toStringEscapeControlChars(value : Dynamic, ?keepStringEscapes : Bool = false) : String {
+		return toStringCommon(value, keepStringEscapes, function(val, s){
+			#if js
+				untyped __js__("
+  					var re = new RegExp(\"[\\\\\\\"\\n\\t\\x00-\\x08\\x0B-\\x1F]\", \"g\");
```
Comment (ykuzmenko73) [inline platforms/common/haxe/HaxeRuntime.hx:373]: why it is in separate variable?

**Review by ykuzmenko73 — COMMENTED** [review]
LGTM

File: platforms/common/haxe/HaxeRuntime.hx (hunk 1 above)
Comment (dmiserg) [reply platforms/common/haxe/HaxeRuntime.hx]: fixed

File: lib/runtime.flow [hunk 2]
```diff
@@ -284,6 +284,20 @@ export {
...
+}
+
+toStringForJson(v : ?) -> string {
+	toStringEscapeControlChars(flow(v));
+}
+
+native toStringEscapeControlChars : (flow) -> string = Native.toStringEscapeControlChars;
+
+toStringEscapeControlChars(v : flow) -> string {
+	println("toStringEscapeControlChars fallback!");
```
Comment (ykuzmenko73) [inline lib/runtime.flow:299]: debug print

**Review by ykuzmenko73 — COMMENTED** [review]
LGTM

File: lib/runtime.flow (hunk 2 above)
Comment (dmiserg) [reply lib/runtime.flow]: fixed

**Pull request merged**